packages/core/src/telemetry/sdk.ts

```diff
@@ -127,33 +128,49 @@ export function initializeTelemetry(config: Config): void {
       exportIntervalMillis: 30000,
     });
   } else if (useOtlp) {
+    const otlpHeaders = config.getTelemetryOtlpHeaders() ?? {};
+    const hasHeaders = Object.keys(otlpHeaders).length > 0;
+
     if (otlpProtocol === 'http') {
       spanExporter = new OTLPTraceExporterHttp({
         url: parsedEndpoint,
+        headers: hasHeaders ? otlpHeaders : undefined,
       });
       logExporter = new OTLPLogExporterHttp({
         url: parsedEndpoint,
+        headers: hasHeaders ? otlpHeaders : undefined,
       });
       metricReader = new PeriodicExportingMetricReader({
         exporter: new OTLPMetricExporterHttp({
           url: parsedEndpoint,
+          headers: hasHeaders ? otlpHeaders : undefined,
         }),
         exportIntervalMillis: 10000,
       });
     } else {
-      // grpc
+      // grpc - convert headers to Metadata
+      let metadata: Metadata | undefined;
+      if (hasHeaders) {
+        metadata = new Metadata();
+        for (const [key, value] of Object.entries(otlpHeaders)) {
+          metadata.set(key, value);
+        }
+      }
```

Comment:
this isn't standards compliant. repeated header keys may be concatenated rather than overwritten.